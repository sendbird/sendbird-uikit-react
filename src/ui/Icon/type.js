export default {
  ADD: 'ADD',
  ARROW_LEFT: 'ARROW_LEFT',
  ATTACH: 'ATTACH',
  BAN: 'BAN',
  BROADCAST: 'BROADCAST',
  CAMERA: 'CAMERA',
  CHANNELS: 'CHANNELS',
  CHAT: 'CHAT',
  CHAT_FILLED: 'CHAT_FILLED',
  CHEVRON_DOWN: 'CHEVRON_DOWN',
  CHEVRON_RIGHT: 'CHEVRON_RIGHT',
  CLOSE: 'CLOSE',
  COLLAPSE: 'COLLAPSE',
  COPY: 'COPY',
  CREATE: 'CREATE',
  DELETE: 'DELETE',
  DISCONNECTED: 'DISCONNECTED',
  DOCUMENT: 'DOCUMENT',
  DONE: 'DONE',
  DONE_ALL: 'DONE_ALL',
  DOWNLOAD: 'DOWNLOAD',
  EDIT: 'EDIT',
  EMOJI_MORE: 'EMOJI_MORE',
  ERROR: 'ERROR',
  EXPAND: 'EXPAND',
  FILE_AUDIO: 'FILE_AUDIO',
  FILE_DOCUMENT: 'FILE_DOCUMENT',
  FREEZE: 'FREEZE',
  GIF: 'GIF',
  INFO: 'INFO',
  LEAVE: 'LEAVE',
  MEMBERS: 'MEMBERS',
  MESSAGE: 'MESSAGE',
  MODERATIONS: 'MODERATIONS',
  MORE: 'MORE',
  MUTE: 'MUTE',
  NOTIFICATIONS: 'NOTIFICATIONS',
  NOTIFICATIONS_OFF_FILLED: 'NOTIFICATIONS_OFF_FILLED',
  OPERATOR: 'OPERATOR',
  PHOTO: 'PHOTO',
  PLAY: 'PLAY',
  PLUS: 'PLUS',
  QUESTION: 'QUESTION',
  REFRESH: 'REFRESH',
  REPLY: 'REPLY',
  REMOVE: 'REMOVE',
  SEARCH: 'SEARCH',
  SEND: 'SEND',
  SETTINGS_FILLED: 'SETTINGS_FILLED',
  SPINNER: 'SPINNER',
  SUPERGROUP: 'SUPERGROUP',
  THUMBNAIL_NONE: 'THUMBNAIL_NONE',
  TOGGLE_OFF: 'TOGGLE_OFF',
  TOGGLE_ON: 'TOGGLE_ON',
  USER: 'USER',
};
