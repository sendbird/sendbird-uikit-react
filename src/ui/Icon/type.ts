/* eslint-disable no-redeclare */
export const Types = {
  ADD: 'ADD',
  ARROW_LEFT: 'ARROW_LEFT',
  ATTACH: 'ATTACH',
  AUDIO_ON_LINED: 'AUDIO_ON_LINED',
  BAN: 'BAN',
  BROADCAST: 'BROADCAST',
  CAMERA: 'CAMERA',
  CHANNELS: 'CHANNELS',
  CHAT: 'CHAT',
  CHAT_FILLED: 'CHAT_FILLED',
  CHEVRON_DOWN: 'CHEVRON_DOWN',
  CHEVRON_RIGHT: 'CHEVRON_RIGHT',
  CLOSE: 'CLOSE',
  COLLAPSE: 'COLLAPSE',
  COPY: 'COPY',
  CREATE: 'CREATE',
  DELETE: 'DELETE',
  DISCONNECTED: 'DISCONNECTED',
  DOCUMENT: 'DOCUMENT',
  DONE: 'DONE',
  DONE_ALL: 'DONE_ALL',
  DOWNLOAD: 'DOWNLOAD',
  EDIT: 'EDIT',
  EMOJI_MORE: 'EMOJI_MORE',
  ERROR: 'ERROR',
  EXPAND: 'EXPAND',
  FILE_AUDIO: 'FILE_AUDIO',
  FILE_DOCUMENT: 'FILE_DOCUMENT',
  FREEZE: 'FREEZE',
  GIF: 'GIF',
  INFO: 'INFO',
  LEAVE: 'LEAVE',
  MEMBERS: 'MEMBERS',
  MESSAGE: 'MESSAGE',
  MODERATIONS: 'MODERATIONS',
  MORE: 'MORE',
  MUTE: 'MUTE',
  NOTIFICATIONS: 'NOTIFICATIONS',
  NOTIFICATIONS_OFF_FILLED: 'NOTIFICATIONS_OFF_FILLED',
  OPERATOR: 'OPERATOR',
  PHOTO: 'PHOTO',
  PLAY: 'PLAY',
  PLUS: 'PLUS',
  QUESTION: 'QUESTION',
  REFRESH: 'REFRESH',
  REPLY: 'REPLY',
  REMOVE: 'REMOVE',
  SEARCH: 'SEARCH',
  SEND: 'SEND',
  SETTINGS_FILLED: 'SETTINGS_FILLED',
  SPINNER: 'SPINNER',
  SUPERGROUP: 'SUPERGROUP',
  THREAD: 'THREAD',
  THUMBNAIL_NONE: 'THUMBNAIL_NONE',
  TOGGLE_OFF: 'TOGGLE_OFF',
  TOGGLE_ON: 'TOGGLE_ON',
  USER: 'USER',
} as const;
export type Types = typeof Types[keyof typeof Types];
