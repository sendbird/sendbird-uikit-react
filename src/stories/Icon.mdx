import { Meta, Canvas, Controls } from '@storybook/addon-docs';
import Icon, { IconTypes, IconColors } from '../ui/Icon';
import * as IconStories from './Icon.stories';

<Meta title="UI/Icon" component={Icon} of={IconStories} />

# Icon

The `Icon` component displays SVG icons. It supports various types and colors, making it versatile for use across the UI.

<Canvas of={IconStories.Default} />
<Controls />

### Importing Icon

To use the `Icon` component in your project, add the following import statement:

```tsx
import Icon, { IconTypes, IconColors } from '@sendbird/uikit-react/ui/Icon';
```
