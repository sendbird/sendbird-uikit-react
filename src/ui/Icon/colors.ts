/* eslint-disable no-redeclare */
export const Colors = {
  DEFAULT: 'DEFAULT',
  PRIMARY: 'PRIMARY',
  PRIMARY_2: 'PRIMARY_2',
  SECONDARY: 'SECONDARY',
  CONTENT: 'CONTENT',
  CONTENT_INVERSE: 'CONTENT_INVERSE',
  CONTENT_INVERSE_5: 'CONTENT_INVERSE_5',
  WHITE: 'WHITE',
  GRAY: 'GRAY',
  THUMBNAIL_ICON: 'THUMBNAIL_ICON',
  SENT: 'SENT',
  READ: 'READ',
  ON_BACKGROUND_1: 'ON_BACKGROUND_1',
  ON_BACKGROUND_2: 'ON_BACKGROUND_2',
  ON_BACKGROUND_3: 'ON_BACKGROUND_3',
  ON_BACKGROUND_4: 'ON_BACKGROUND_4',
  BACKGROUND_3: 'BACKGROUND_3',
  ERROR: 'ERROR',
} as const;
export type Colors = typeof Colors[keyof typeof Colors];

export default { Colors };
