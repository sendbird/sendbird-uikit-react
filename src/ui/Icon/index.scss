@import '../../styles/variables';

.sendbird-icon {
  display: inline-block;
  &:focus {
    outline: none;
  }
}

.sendbird-icon-color--primary [class*='fill'] {
  @include themed() {
    fill: t(primary-3);
  }
}
.sendbird-icon-color--primary-2 [class*='fill'] {
  @include themed() {
    fill: t(primary-2);
  }
}

.sendbird-icon-color--primary [class*='stroke'] {
  @include themed() {
    stroke: t(primary-3);
    stroke-width: 2px;
    stroke-linejoin: round;
  }
}

.sendbird-icon-color--secondary [class*='fill'] {
  @include themed() {
    fill: t(secondary-3);
  }
}

.sendbird-icon-color--content [class*='fill'] {
  @include themed() {
    fill: t(on-content-1);
  }
}

.sendbird-icon-color--content-inverse [class*='fill'] {
  @include themed() {
    fill: t(inverse-on-content-1);
  }
}

.sendbird-icon-color--content-inverse-5 [class*='fill'] {
  @include themed() {
    fill: t(inverse-on-content-5);
  }
}

.sendbird-icon-color--white [class*='fill'] {
  fill: #ffffff;
}

.sendbird-icon-color--gray [class*='fill'] {
  fill: var(--sendbird-light-onlight-04);
}

.sendbird-icon-color--thumbnail-icon [class*='fill'] {
  fill: var(--sendbird-light-onlight-02);
}

.sendbird-icon-color--sent [class*='fill'] {
  @include themed() {
    fill: t(inverse-on-content-3);
  }
}

.sendbird-icon-color--read [class*='fill'] {
  @include themed() {
    fill: t(secondary--3-2);
  }
}

.sendbird-icon-color--on-background-1 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-1);
  }
}
.sendbird-icon-color--on-background-2 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-2);
  }
}
.sendbird-icon-color--on-background-3 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-3);
  }
}
.sendbird-icon-color--on-background-4 [class*='fill'] {
  @include themed() {
    fill: t(on-bg-4);
  }
}

.sendbird-icon-color--background-3 [class*='fill'] {
  @include themed() {
    fill: t(bg-3);
  }
}

.sendbird-icon-color--error [class*='fill'] {
  @include themed() {
    fill: t(error--3-2);
  }
}
