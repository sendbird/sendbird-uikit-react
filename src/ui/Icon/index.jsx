import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';
import { Types } from './type';
import { Colors } from './colors';
import {
  changeColorToClassName,
  changeTypeToIconClassName,
} from './utils';

import IconAdd from '../../svgs/icon-add.svg';
import IconArrowLeft from '../../svgs/icon-arrow-left.svg';
import IconAttach from '../../svgs/icon-attach.svg';
import IconAudioOnLined from '../../svgs/icon-audio-on-lined.svg';
import IconBan from '../../svgs/icon-ban.svg';
import IconBroadcast from '../../svgs/icon-broadcast.svg';
import IconCamera from '../../svgs/icon-camera.svg';
import IconChannels from '../../svgs/icon-channels.svg';
import IconChat from '../../svgs/icon-chat.svg';
import IconChatFilled from '../../svgs/icon-chat-filled.svg';
import IconChevronDown from '../../svgs/icon-chevron-down.svg';
import IconChevronRight from '../../svgs/icon-chevron-right.svg';
import IconClose from '../../svgs/icon-close.svg';
import IconCollapse from '../../svgs/icon-collapse.svg';
import IconCopy from '../../svgs/icon-copy.svg';
import IconCreate from '../../svgs/icon-create.svg';
import IconDelete from '../../svgs/icon-delete.svg';
import IconDisconnected from '../../svgs/icon-disconnected.svg';
import IconDocument from '../../svgs/icon-document.svg';
import IconDone from '../../svgs/icon-done.svg';
import IconDoneAll from '../../svgs/icon-done-all.svg';
import IconDownload from '../../svgs/icon-download.svg';
import IconEdit from '../../svgs/icon-edit.svg';
import IconEmojiMore from '../../svgs/icon-emoji-more.svg';
import IconError from '../../svgs/icon-error.svg';
import IconExpand from '../../svgs/icon-expand.svg';
import IconFileAudio from '../../svgs/icon-file-audio.svg';
import IconFileDocument from '../../svgs/icon-file-document.svg';
import IconFreeze from '../../svgs/icon-freeze.svg';
import IconGif from '../../svgs/icon-gif.svg';
import IconInfo from '../../svgs/icon-info.svg';
import IconLeave from '../../svgs/icon-leave.svg';
import IconMembers from '../../svgs/icon-members.svg';
import IconMessage from '../../svgs/icon-message.svg';
import IconModerations from '../../svgs/icon-moderations.svg';
import IconMore from '../../svgs/icon-more.svg';
import IconMute from '../../svgs/icon-mute.svg';
import IconNotifications from '../../svgs/icon-notifications.svg';
import IconNotificationsOffFilled from '../../svgs/icon-notifications-off-filled.svg';
import IconOperator from '../../svgs/icon-operator.svg';
import IconPhoto from '../../svgs/icon-photo.svg';
import IconPlay from '../../svgs/icon-play.svg';
import IconPlus from '../../svgs/icon-plus.svg';
import IconQuestion from '../../svgs/icon-question.svg';
import IconRefresh from '../../svgs/icon-refresh.svg';
import IconRemove from '../../svgs/icon-remove.svg';
import IconReply from '../../svgs/icon-reply-filled.svg';
import IconSearch from '../../svgs/icon-search.svg';
import IconSend from '../../svgs/icon-send.svg';
import IconSettingsFilled from '../../svgs/icon-settings-filled.svg';
import IconSpinner from '../../svgs/icon-spinner.svg';
import IconSupergroup from '../../svgs/icon-supergroup.svg';
import IconThread from '../../svgs/icon-thread.svg';
import IconThumbnailNone from '../../svgs/icon-thumbnail-none.svg';
import IconToggleOff from '../../svgs/icon-toggleoff.svg';
import IconToggleOn from '../../svgs/icon-toggleon.svg';
import IconUser from '../../svgs/icon-user.svg';

function changeTypeToIconComponent(type) {
  switch (type) {
    case Types.ADD: return <IconAdd />;
    case Types.ARROW_LEFT: return <IconArrowLeft />;
    case Types.ATTACH: return <IconAttach />;
    case Types.AUDIO_ON_LINED: return <IconAudioOnLined />;
    case Types.BAN: return <IconBan />;
    case Types.BROADCAST: return <IconBroadcast />;
    case Types.CAMERA: return <IconCamera />;
    case Types.CHANNELS: return <IconChannels />;
    case Types.CHAT: return <IconChat />;
    case Types.CHAT_FILLED: return <IconChatFilled />;
    case Types.CHEVRON_DOWN: return <IconChevronDown />;
    case Types.CHEVRON_RIGHT: return <IconChevronRight />;
    case Types.CLOSE: return <IconClose />;
    case Types.COLLAPSE: return <IconCollapse />;
    case Types.COPY: return <IconCopy />;
    case Types.CREATE: return <IconCreate />;
    case Types.DELETE: return <IconDelete />;
    case Types.DISCONNECTED: return <IconDisconnected />;
    case Types.DOCUMENT: return <IconDocument />;
    case Types.DONE: return <IconDone />;
    case Types.DONE_ALL: return <IconDoneAll />;
    case Types.DOWNLOAD: return <IconDownload />;
    case Types.EDIT: return <IconEdit />;
    case Types.EMOJI_MORE: return <IconEmojiMore />;
    case Types.ERROR: return <IconError />;
    case Types.EXPAND: return <IconExpand />;
    case Types.FILE_AUDIO: return <IconFileAudio />;
    case Types.FILE_DOCUMENT: return <IconFileDocument />;
    case Types.FREEZE: return <IconFreeze />;
    case Types.GIF: return <IconGif />;
    case Types.INFO: return <IconInfo />;
    case Types.LEAVE: return <IconLeave />;
    case Types.MEMBERS: return <IconMembers />;
    case Types.MESSAGE: return <IconMessage />;
    case Types.MODERATIONS: return <IconModerations />;
    case Types.MORE: return <IconMore />;
    case Types.MUTE: return <IconMute />;
    case Types.NOTIFICATIONS: return <IconNotifications />;
    case Types.NOTIFICATIONS_OFF_FILLED: return <IconNotificationsOffFilled />;
    case Types.OPERATOR: return <IconOperator />;
    case Types.PHOTO: return <IconPhoto />;
    case Types.PLAY: return <IconPlay />;
    case Types.PLUS: return <IconPlus />;
    case Types.QUESTION: return <IconQuestion />;
    case Types.REFRESH: return <IconRefresh />;
    case Types.REMOVE: return <IconRemove />;
    case Types.REPLY: return <IconReply />;
    case Types.SEARCH: return <IconSearch />;
    case Types.SEND: return <IconSend />;
    case Types.SETTINGS_FILLED: return <IconSettingsFilled />;
    case Types.SPINNER: return <IconSpinner />;
    case Types.SUPERGROUP: return <IconSupergroup />;
    case Types.THREAD: return <IconThread />;
    case Types.THUMBNAIL_NONE: return <IconThumbnailNone />;
    case Types.TOGGLE_OFF: return <IconToggleOff />;
    case Types.TOGGLE_ON: return <IconToggleOn />;
    case Types.USER: return <IconUser />;
    default: return 'icon'; // If you see this text 'icon' replace icon for it
  }
}

export default function Icon({
  className,
  type,
  fillColor,
  width,
  height,
  onClick,
  children,
}) {
  const iconStyle = {
    width: typeof width === 'string' ? width : `${width}px`,
    minWidth: typeof width === 'string' ? width : `${width}px`,
    height: typeof height === 'string' ? height : `${height}px`,
    minHeight: typeof height === 'string' ? height : `${height}px`,
  };
  return (
    <div
      className={[
        ...Array.isArray(className) ? className : [className],
        'sendbird-icon',
        changeTypeToIconClassName(type),
        changeColorToClassName(fillColor),
      ].join(' ')}
      role="button"
      onClick={onClick}
      onKeyDown={onClick}
      tabIndex="0"
      style={iconStyle}
    >
      {children || changeTypeToIconComponent(type)}
    </div>
  );
}

Icon.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  /** Type: Use strings from below list */
  type: PropTypes.oneOfType([
    PropTypes.oneOf(Object.keys(Types)),
    PropTypes.string,
  ]).isRequired,
  /** Type: Use Colors from below list */
  fillColor: PropTypes.oneOf(Object.keys(Colors)),
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onClick: PropTypes.func,
  children: PropTypes.element,
};

Icon.defaultProps = {
  className: '',
  fillColor: Colors.DEFAULT,
  width: 26,
  height: 26,
  onClick: () => { },
  children: null,
};

export const IconTypes = Types;
export const IconColors = Colors;
