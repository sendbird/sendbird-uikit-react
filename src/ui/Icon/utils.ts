import { Types } from './type';
import { Colors } from './colors';

export function changeColorToClassName(color: Colors): string {
  switch (color) {
    case Colors.PRIMARY: return 'sendbird-icon-color--primary';
    case Colors.PRIMARY_2: return 'sendbird-icon-color--primary-2';
    case Colors.SECONDARY: return 'sendbird-icon-color--secondary';
    case Colors.CONTENT: return 'sendbird-icon-color--content';
    case Colors.CONTENT_INVERSE: return 'sendbird-icon-color--content-inverse';
    case Colors.WHITE: return 'sendbird-icon-color--white';
    case Colors.GRAY: return 'sendbird-icon-color--gray';
    case Colors.SENT: return 'sendbird-icon-color--sent';
    case Colors.READ: return 'sendbird-icon-color--read';
    case Colors.ON_BACKGROUND_1: return 'sendbird-icon-color--on-background-1';
    case Colors.ON_BACKGROUND_2: return 'sendbird-icon-color--on-background-2';
    case Colors.ON_BACKGROUND_3: return 'sendbird-icon-color--on-background-3';
    case Colors.ON_BACKGROUND_4: return 'sendbird-icon-color--on-background-4';
    case Colors.BACKGROUND_3: return 'sendbird-icon-color--background-3';
    case Colors.ERROR: return 'sendbird-icon-color--error';
    default: return '';
  }
}

export function changeTypeToIconClassName(type: Types): string {
  switch (type) {
    case Types.ADD: return 'sendbird-icon-add';
    case Types.ARROW_LEFT: return 'sendbird-icon-arrow-left';
    case Types.ATTACH: return 'sendbird-icon-attach';
    case Types.AUDIO_ON_LINED: return 'sendbird-icon-audio-on-lined';
    case Types.BAN: return 'sendbird-icon-ban';
    case Types.BROADCAST: return 'sendbird-icon-broadcast';
    case Types.CAMERA: return 'sendbird-icon-camera';
    case Types.CHANNELS: return 'sendbird-icon-channels';
    case Types.CHAT: return 'sendbird-icon-chat';
    case Types.CHAT_FILLED: return 'sendbird-icon-chat-filled';
    case Types.CHEVRON_DOWN: return 'sendbird-icon-chevron-down';
    case Types.CHEVRON_RIGHT: return 'sendbird-icon-chevron-right';
    case Types.CLOSE: return 'sendbird-icon-close';
    case Types.COLLAPSE: return 'sendbird-icon-collapse';
    case Types.COPY: return 'sendbird-icon-copy';
    case Types.CREATE: return 'sendbird-icon-create';
    case Types.DELETE: return 'sendbird-icon-delete';
    case Types.DISCONNECTED: return 'sendbird-icon-disconnected';
    case Types.DOCUMENT: return 'sendbird-icon-document';
    case Types.DONE: return 'sendbird-icon-done';
    case Types.DONE_ALL: return 'sendbird-icon-done-all';
    case Types.DOWNLOAD: return 'sendbird-icon-down-load';
    case Types.EDIT: return 'sendbird-icon-edit';
    case Types.EMOJI_MORE: return 'sendbird-icon-emoji-more';
    case Types.ERROR: return 'sendbird-icon-error';
    case Types.EXPAND: return 'sendbird-icon-expand';
    case Types.FILE_AUDIO: return 'sendbird-icon-file-audio';
    case Types.FILE_DOCUMENT: return 'sendbird-icon-file-document';
    case Types.FREEZE: return 'sendbird-icon-freeze';
    case Types.GIF: return 'sendbird-icon-gif';
    case Types.INFO: return 'sendbird-icon-info';
    case Types.LEAVE: return 'sendbird-icon-leave';
    case Types.MEMBERS: return 'sendbird-icon-members';
    case Types.MESSAGE: return 'sendbird-icon-message';
    case Types.MODERATIONS: return 'sendbird-icon-moderations';
    case Types.MORE: return 'sendbird-icon-more';
    case Types.MUTE: return 'sendbird-icon-mute';
    case Types.NOTIFICATIONS: return 'sendbird-icon-notifications';
    case Types.NOTIFICATIONS_OFF_FILLED: return 'sendbird-icon-notifications-off-filled';
    case Types.OPERATOR: return 'sendbird-icon-operator';
    case Types.PHOTO: return 'sendbird-icon-photo';
    case Types.PLAY: return 'sendbird-icon-play';
    case Types.PLUS: return 'sendbird-iconn-plus';
    case Types.QUESTION: return 'sendbird-icon-question';
    case Types.REFRESH: return 'sendbird-icon-refresh';
    case Types.REMOVE: return 'sendbird-icon-remove';
    case Types.REPLY: return 'sendbird-icon-reply';
    case Types.SEARCH: return 'sendbird-icon-search';
    case Types.SEND: return 'sendbird-icon-send';
    case Types.SETTINGS_FILLED: return 'sendbird-icon-settings-filled';
    case Types.SPINNER: return 'sendbird-icon-spinner';
    case Types.SUPERGROUP: return 'sendbird-icon-supergroup';
    case Types.THREAD: return 'sendbird-icon-thread';
    case Types.THUMBNAIL_NONE: return 'sendbird-icon-thumbnail-none';
    case Types.TOGGLE_OFF: return 'sendbird-icon-toggle-off';
    case Types.TOGGLE_ON: return 'sendbird-icon-toggle-on';
    case Types.USER: return 'sendbird-icon-user';
    default: return 'sendbird-icon-unknown'; // If you see this text 'icon' replace icon for it
  }
}
