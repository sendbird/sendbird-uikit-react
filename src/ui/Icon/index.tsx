import React from 'react';

import './index.scss';
import { Types } from './type';
import { Colors } from './colors';
import {
  changeColorToClassName,
  changeTypeToIconClassName,
} from './utils';

import IconAdd from '../../svgs/icon-add.svg';
import IconArrowLeft from '../../svgs/icon-arrow-left.svg';
import IconAttach from '../../svgs/icon-attach.svg';
import IconAudioOnLined from '../../svgs/icon-audio-on-lined.svg';
import IconBan from '../../svgs/icon-ban.svg';
import IconBroadcast from '../../svgs/icon-broadcast.svg';
import IconCamera from '../../svgs/icon-camera.svg';
import IconChannels from '../../svgs/icon-channels.svg';
import IconChat from '../../svgs/icon-chat.svg';
import IconChatFilled from '../../svgs/icon-chat-filled.svg';
import IconChevronDown from '../../svgs/icon-chevron-down.svg';
import IconChevronRight from '../../svgs/icon-chevron-right.svg';
import IconClose from '../../svgs/icon-close.svg';
import IconCollapse from '../../svgs/icon-collapse.svg';
import IconCopy from '../../svgs/icon-copy.svg';
import IconCreate from '../../svgs/icon-create.svg';
import IconDelete from '../../svgs/icon-delete.svg';
import IconDisconnected from '../../svgs/icon-disconnected.svg';
import IconDocument from '../../svgs/icon-document.svg';
import IconDone from '../../svgs/icon-done.svg';
import IconDoneAll from '../../svgs/icon-done-all.svg';
import IconDownload from '../../svgs/icon-download.svg';
import IconEdit from '../../svgs/icon-edit.svg';
import IconEmojiMore from '../../svgs/icon-emoji-more.svg';
import IconError from '../../svgs/icon-error.svg';
import IconExpand from '../../svgs/icon-expand.svg';
import IconFileAudio from '../../svgs/icon-file-audio.svg';
import IconFileDocument from '../../svgs/icon-file-document.svg';
import IconFreeze from '../../svgs/icon-freeze.svg';
import IconGif from '../../svgs/icon-gif.svg';
import IconInfo from '../../svgs/icon-info.svg';
import IconLeave from '../../svgs/icon-leave.svg';
import IconMembers from '../../svgs/icon-members.svg';
import IconMessage from '../../svgs/icon-message.svg';
import IconModerations from '../../svgs/icon-moderations.svg';
import IconMore from '../../svgs/icon-more.svg';
import IconMute from '../../svgs/icon-mute.svg';
import IconNotifications from '../../svgs/icon-notifications.svg';
import IconNotificationsOffFilled from '../../svgs/icon-notifications-off-filled.svg';
import IconOperator from '../../svgs/icon-operator.svg';
import IconPhoto from '../../svgs/icon-photo.svg';
import IconPlay from '../../svgs/icon-play.svg';
import IconPlus from '../../svgs/icon-plus.svg';
import IconQuestion from '../../svgs/icon-question.svg';
import IconRefresh from '../../svgs/icon-refresh.svg';
import IconRemove from '../../svgs/icon-remove.svg';
import IconReply from '../../svgs/icon-reply-filled.svg';
import IconSearch from '../../svgs/icon-search.svg';
import IconSend from '../../svgs/icon-send.svg';
import IconSettingsFilled from '../../svgs/icon-settings-filled.svg';
import IconSlideLeft from '../../svgs/icon-slide-left.svg';
import IconSpinner from '../../svgs/icon-spinner.svg';
import IconSupergroup from '../../svgs/icon-supergroup.svg';
import IconThread from '../../svgs/icon-thread.svg';
import IconThumbnailNone from '../../svgs/icon-thumbnail-none.svg';
import IconToggleOff from '../../svgs/icon-toggleoff.svg';
import IconToggleOn from '../../svgs/icon-toggleon.svg';
import IconUser from '../../svgs/icon-user.svg';
import IconFeedbackLike from '../../svgs/icon-feedback-like.svg';
import IconFeedbackDislike from '../../svgs/icon-feedback-dislike.svg';

import { noop } from '../../utils/utils';

function changeTypeToIconComponent(type: Types) {
  switch (type) {
    case Types.ADD: return <IconAdd />;
    case Types.ARROW_LEFT: return <IconArrowLeft />;
    case Types.ATTACH: return <IconAttach />;
    case Types.AUDIO_ON_LINED: return <IconAudioOnLined />;
    case Types.BAN: return <IconBan />;
    case Types.BROADCAST: return <IconBroadcast />;
    case Types.CAMERA: return <IconCamera />;
    case Types.CHANNELS: return <IconChannels />;
    case Types.CHAT: return <IconChat />;
    case Types.CHAT_FILLED: return <IconChatFilled />;
    case Types.CHEVRON_DOWN: return <IconChevronDown />;
    case Types.CHEVRON_RIGHT: return <IconChevronRight />;
    case Types.CLOSE: return <IconClose />;
    case Types.COLLAPSE: return <IconCollapse />;
    case Types.COPY: return <IconCopy />;
    case Types.CREATE: return <IconCreate />;
    case Types.DELETE: return <IconDelete />;
    case Types.DISCONNECTED: return <IconDisconnected />;
    case Types.DOCUMENT: return <IconDocument />;
    case Types.DONE: return <IconDone />;
    case Types.DONE_ALL: return <IconDoneAll />;
    case Types.DOWNLOAD: return <IconDownload />;
    case Types.EDIT: return <IconEdit />;
    case Types.EMOJI_MORE: return <IconEmojiMore />;
    case Types.ERROR: return <IconError />;
    case Types.EXPAND: return <IconExpand />;
    case Types.FILE_AUDIO: return <IconFileAudio />;
    case Types.FILE_DOCUMENT: return <IconFileDocument />;
    case Types.FREEZE: return <IconFreeze />;
    case Types.GIF: return <IconGif />;
    case Types.INFO: return <IconInfo />;
    case Types.LEAVE: return <IconLeave />;
    case Types.MEMBERS: return <IconMembers />;
    case Types.MESSAGE: return <IconMessage />;
    case Types.MODERATIONS: return <IconModerations />;
    case Types.MORE: return <IconMore />;
    case Types.MUTE: return <IconMute />;
    case Types.NOTIFICATIONS: return <IconNotifications />;
    case Types.NOTIFICATIONS_OFF_FILLED: return <IconNotificationsOffFilled />;
    case Types.OPERATOR: return <IconOperator />;
    case Types.PHOTO: return <IconPhoto />;
    case Types.PLAY: return <IconPlay />;
    case Types.PLUS: return <IconPlus />;
    case Types.QUESTION: return <IconQuestion />;
    case Types.REFRESH: return <IconRefresh />;
    case Types.REMOVE: return <IconRemove />;
    case Types.REPLY: return <IconReply />;
    case Types.SEARCH: return <IconSearch />;
    case Types.SEND: return <IconSend />;
    case Types.SETTINGS_FILLED: return <IconSettingsFilled />;
    case Types.SLIDE_LEFT: return <IconSlideLeft />;
    case Types.SPINNER: return <IconSpinner />;
    case Types.SUPERGROUP: return <IconSupergroup />;
    case Types.THREAD: return <IconThread />;
    case Types.THUMBNAIL_NONE: return <IconThumbnailNone />;
    case Types.TOGGLE_OFF: return <IconToggleOff />;
    case Types.TOGGLE_ON: return <IconToggleOn />;
    case Types.USER: return <IconUser />;
    case Types.FEEDBACK_LIKE: return <IconFeedbackLike />;
    case Types.FEEDBACK_DISLIKE: return <IconFeedbackDislike />;
    default: return 'icon'; // If you see this text 'icon' replace icon for it
  }
}

type IconProps = {
  className?: string | string[];
  /** Type: Use strings from below list */
  type: Types;
  /** Type: Use Colors from below list */
  fillColor?: Colors;
  width?: string | number;
  height?: string | number;
  onClick?: React.MouseEventHandler<HTMLDivElement> & React.KeyboardEventHandler<HTMLDivElement>;
  children?: React.ReactNode;
};
export default function Icon({
  className = '',
  type,
  fillColor = Colors.DEFAULT,
  width = 26,
  height = 26,
  onClick = noop,
  children = null,
}: IconProps) {
  const iconStyle = {
    width: typeof width === 'string' ? width : `${width}px`,
    minWidth: typeof width === 'string' ? width : `${width}px`,
    height: typeof height === 'string' ? height : `${height}px`,
    minHeight: typeof height === 'string' ? height : `${height}px`,
  };
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-icon',
        changeTypeToIconClassName(type),
        changeColorToClassName(fillColor),
      ].join(' ')}
      role={'button'}
      onClick={onClick}
      onKeyDown={onClick}
      tabIndex={0}
      style={iconStyle}
    >
      {children || changeTypeToIconComponent(type)}
    </div>
  );
}

export const IconTypes = Types;
export const IconColors = Colors;
