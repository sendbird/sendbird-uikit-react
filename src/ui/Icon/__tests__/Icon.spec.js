import React from 'react';
import { render } from '@testing-library/react';

import Icon, { IconTypes } from "../index";

describe('ui/Icon', () => {
  it('should do a snapshot test of the defualt Icon DOM', function () {
    const { asFragment } = render(
      <Icon type={IconTypes.ADD} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
