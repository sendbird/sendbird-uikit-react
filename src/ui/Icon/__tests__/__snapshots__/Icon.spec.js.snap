// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Icon should do a snapshot test of the defualt Icon DOM 1`] = `
<DocumentFragment>
  <div
    class=" sendbird-icon sendbird-icon-add "
    role="button"
    style="width: 26px; min-width: 26px; height: 26px; min-height: 26px;"
    tabindex="0"
  >
    <test-file-stub />
  </div>
</DocumentFragment>
`;
